### PR TITLE
[JENKINS-54128] Deprecated calls to Run.getLogFile

### PR DESCRIPTION
**Pull request by basil**
See [JENKINS-54128](https://issues.jenkins-ci.org/browse/JENKINS-54128). Removes this plugin's usage of the deprecated `Run#getLogFile` method, which has been deprecated as of [JEP-210](https://github.com/jenkinsci/jep/tree/master/jep/210), in favor of the non-deprecated `Run#getLogReader` method. The latter internally handles character set issues, so there is also no need to deal with the character set when calling the method. As a bonus, I also switched to the Java 7 try-with-resources syntax.

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/tokenmacro/impl/LogRegExMacro.java [hunk 1]
```diff
@@ -65,40 +52,26 @@ public String evaluate(AbstractBuild<?, ?> context, TaskListener listener, Strin
 
     @Override
     public String evaluate(Run<?, ?> run, FilePath workspace, TaskListener listener, String macroName) throws MacroEvaluationException, IOException, InterruptedException {
-        return readLogFile(run.getLogFile());
+        return readLogFile(run);
     }
 
-    public String readLogFile(File file) throws IOException {
+    private String readLogFile(Run<?, ?> run) throws IOException {
```
Comment (varyvol) [inline src/main/java/org/jenkinsci/plugins/tokenmacro/impl/LogRegExMacro.java]: Isn't changing the visibility a bit risky?

Comment (varyvol) [reply src/main/java/org/jenkinsci/plugins/tokenmacro/impl/LogRegExMacro.java]: @basil @slide

Comment (slide) [reply src/main/java/org/jenkinsci/plugins/tokenmacro/impl/LogRegExMacro.java]: It's a bit risky, yes. I am fine with it though

Comment (varyvol) [reply src/main/java/org/jenkinsci/plugins/tokenmacro/impl/LogRegExMacro.java]: Fine then, thanks!

**Pull request merged**